### PR TITLE
feat: update private interface to generate tools meta

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

+ include refereed packages information
+ support `custom_llm` node

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_utils.py [hunk 1]
```diff
@@ -603,7 +604,7 @@ def _generate_package_tools() -> dict:
 
     from promptflow._core.tools_manager import collect_package_tools
 
-    return collect_package_tools()
+    return collect_package_tools(keys=keys)
 
 
 def generate_flow_tools_json(
```
Comment (D-W-) [inline src/promptflow/promptflow/_sdk/_utils.py:610]: What's the difference between this and `generate_tool_meta_dict_by_file`?

File: src/promptflow/promptflow/_core/tool_meta_generator.py [hunk 2]
```diff
@@ -239,6 +239,8 @@ def generate_tool_meta_dict_by_file(path: str, tool_type: ToolType):
         return generate_prompt_meta_dict(name, content, source=path)
     elif tool_type == ToolType.PROMPT:
         return generate_prompt_meta_dict(name, content, prompt_only=True, source=path)
+    elif tool_type == ToolType.CUSTOM_LLM:
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/_core/tool_meta_generator.py:242]: not sure if we also need to include executor guys to aware this change

File: src/promptflow/promptflow/_sdk/_utils.py (hunk 1 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_utils.py:610]: Just saw the `_generate_metas_from_files` func call

File: src/promptflow/promptflow/_core/tool_meta_generator.py (hunk 2 above)
Comment (elliotzh) [reply src/promptflow/promptflow/_core/tool_meta_generator.py:242]: This type has already been added in constant class and I haven't change existing logic, so I think it's safe and just mean that we have no test on this before hh.

cc @thy09 for this. CustomLLM is not supported in tools meta generation before, any concern on updating this?

**Review by D-W- — APPROVED** [review]

**Review by zhengfeiwang — DISMISSED** [review]

File: src/promptflow/promptflow/_core/tool_meta_generator.py (hunk 2 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_core/tool_meta_generator.py:242]: approved and leave this comment open, once we confirm with Heyi, I think we can merge this.

Comment (elliotzh) [reply src/promptflow/promptflow/_core/tool_meta_generator.py]: as discussed, I have generate `package_with_prompt` as `prompt` for now. will raise new pull request if we need to remove `type` and `source` in `flow.tools.json`

**Review by zhengfeiwang — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**